packages/ts/generator-typescript-plugin-model/src/ValidationConstraintParser.ts

```diff
@@ -13,51 +13,45 @@ import ts, {
 } from 'typescript';
 
 export type ValidationConstrainedSchema = NonComposedRegularSchema &
-  Readonly<{ 'x-validation-constraints': readonly Annotation[] }>;
+  Readonly<{ 'x-validation-constraints': readonly ValidationConstraint[] }>;
 
-export function isValidationConstrainedSchema(schema: Schema): schema is ValidationConstrainedSchema {
-  return isNonComposedRegularSchema(schema) && 'x-validation-constraints' in schema;
+export function hasValidationConstraints(schema: Schema): schema is ValidationConstrainedSchema {
+  return (
+    isNonComposedRegularSchema(schema) &&
+    'x-validation-constraints' in schema &&
+    (schema as ValidationConstrainedSchema)['x-validation-constraints'].length > 0
+  );
 }
 
-export interface Annotation {
+export interface ValidationConstraint {
   simpleName: string;
   attributes?: Record<string, unknown>;
 }
 
-export type AnnotationImporter = (name: string) => Identifier;
+export type ValidationConstraintImporter = (name: string) => Identifier;
 
 function selector<T extends Expression>([statement]: readonly Statement[]): T {
   return (statement as VariableStatement).declarationList.declarations[0].initializer as T;
 }
 
 const variableStatementVar = 'const a';
 
-export class AnnotationParser {
-  readonly #importer: AnnotationImporter;
+export class ValidationConstraintParser {
+  readonly #importer: ValidationConstraintImporter;
 
-  constructor(importer: AnnotationImporter) {
+  constructor(importer: ValidationConstraintImporter) {
     this.#importer = importer;
   }
 
-  parse(annotation: Annotation | string): ts.NewExpression {
-    if (typeof annotation === 'string') {
-      const nameEndIndex = annotation.indexOf('(');
-      const simpleName = nameEndIndex >= 0 ? annotation.slice(0, nameEndIndex) : annotation;
-      const id = this.#importer(simpleName);
-
-      return template<NewExpression>(`${variableStatementVar} = new ${annotation}`, selector, [
-        transform((node) => (ts.isIdentifier(node) && node.text === simpleName ? id : node)),
-      ]);
-    }
```

Comment:
This is some left-over legacy code related to what has been removed in https://github.com/vaadin/hilla/pull/1315